### ISSUE TITLE
Enhancement: System font as default, Roboto as fallback

### ISSUE THREAD

**Issue by 5had0w5talk3r** [FreeTubeApp/FreeTube#39]
Using the system sans-serif font as a default would provide better integration than keeping it as fallback.

**Comment by PrestonN** [comment]
I see no problem with doing this, however I think I'd rather have it the other way around where Roboto is the default and there would be a setting to switch to the system defaults.  Only looking through very quick research, this looks like it can go a step further and allow the user to pick any font that they have installed on their system.  I'll keep this is mind but for now I'd consider this low priority.

Thank you for the suggestions :)

**Comment by 5had0w5talk3r** [comment]
Shouldn't integration with the user's system take precedent for a desktop application? Roboto's a nice font, sure; but if the user really likes Roboto, they can set it as their default font without much trouble, or even set an environment variable just for FreeTube.

**Comment by PrestonN** [comment]
I'd be interested in getting a couple of opinions on this.  I'm don't have a very strong opinion on either direction we go with this.  I believe that each direction has their own benefits.

Defaulting to Roboto (or any font set by the developer) helps keep a defined look across all platforms.  This keeps it so that you know how the application will look no matter which platform you choose.

Defaulting to the system font on the platform allows the user to have their application stack more uniform to their tastes, assuming they have a unique font that they've set.  You have to remember that with Windows, (which roughly account for half of the downloads for FreeTube) have a hard time customizing the fonts on their system.  It's possible, but not exactly as easy as it is on Linux based systems.  Some users may not have that same sense of design that you or someone else does and would rather trust the developers / designers to pick out things such as font for them.  It's a very subjective situation but it's highly possible.

Like I've mentioned, I'm okay with this going either direction.  I'm just curious as to what others suggest as far as their preference goes.

**Comment by 5had0w5talk3r** [comment]
I'm personally not keen on the notion of "having a defined look across platforms". To me, it's always stunk of egocentrism of developers who go on and on about "their vision". You do raise a valid point about Windows, however. Their default font is kind of rubish and most applications do their own thing over there, anyway.

However - seeing as MacOS' system fonts are much nicer, and Linux users typically prefer to have their applications follow a uniform look - maybe there could be a different build path for the Windows version where sets Roboto as default and leaves the system default as fallback, and then another for MacOS and Linux, where the opposite happens.

More opinions would be nice, but seeing as the project is in its infancy, I'm not too sure where you'd go to get them. Maybe set up a strawpoll on the releases page?

**Comment by linuxgirl22** [comment]
Hello :-)
I just read the master list and clicked through to here. Sounds great about sorting fonts; I just looked up Roboto and it looks nice. I'd mentioned in another thread, about Vue, that the small fonts on the side bar and under videos (channel name), and the comments, are too small to read, and Preston had said to mention this after the first release of Vue, and he could sort out UI scaling. I just saw that Vue has gone from the Aur and the github page for Vue is read only, so am presuming that Vue has integrated into here? ... it's okay to tag on about the UI scaling here? If not, my apologies.

**Comment by PrestonN** [comment]
Yes, development is coming back to here and the Vue repository has been archived.

I'd say UI scaling should be a separate issue, as it's slightly different than choosing the font within the app, so let's go with that.

**Comment by linuxgirl22** [comment]
That's great; no problem. :-) Do you want me to start a new thread now, or wait?

**Comment by PrestonN** [comment]
You can make a new thread now if you'd like :)

**Comment by linuxgirl22** [comment]
Thanks :-) Will do that now.

**Comment by Golddouble** [comment]
Please at least make it possible to choose in settings between an antialiasing friendly font and a "antialiasing free" friendly font (like Arial). 
https://github.com/FreeTubeApp/FreeTube/issues/3063

**Comment by Golddouble** [comment]
As I use an antialiasing free Linux, the FreeTube font looks very ugly:
![k20230120-125728](https://user-images.githubusercontent.com/13977359/213689937-44651e25-7540-4b00-bc07-83fbda838c17.png)
(-> Please click on the picture to see it with a scale of 100% to see, what I see)

I would appreciate it if issue #39 would be implemented.

**Comment by Golddouble** [comment]
Is there a workaround to enable antialiasing only for the flatpak FreeTube, when I start it. Has FreeTube for example a environmental variable that can be set by start, to enable antialiasing only for Freetube?

Thank you.

**Comment by golden-peaches** [comment]
Hello,

I wanted to resuggest the idea of having FreeTube default to the system's font (at least on linux). The Roboto font does not work well with Japanese. The application defaults to a font for Japanese characters that does not fit well with the Latin Roboto characters. The Japanese is therefore hard to read.

![Screenshot_2023-05-04_08:00:54_JST](https://user-images.githubusercontent.com/120830230/236069315-a2be9581-4dcd-4b26-b311-9806b79babea.png)

**Comment by Hund** [comment]
Any news on this? This is seriously something that keeps me from using this client. I can't stand the default font. It's horrible on my screen. :)

**Comment by Erick12890** [comment]
I hope you are well. I noticed that my recent feature suggestion regarding custom font integration was rejected, and I wanted to follow up on this. I believe that allowing users to personalize the font in FreeTube, particularly by uploading custom .ttf files, could significantly enhance the program's visual appeal and customization options for a wider user base.

While I understand that not all suggestions may fit within your development roadmap, I encourage a more democratic approach when considering community feedback. Features that allow users to tailor their experience often lead to more engagement and satisfaction, ultimately enriching the program for everyone.

If it is not possible to include this feature at the moment, could you please clarify how we can properly suggest and request new features in a way that aligns with FreeTube’s development process? I would appreciate any guidance you can provide to ensure that future suggestions are considered more thoroughly.

Thank you for your time and for the work your team does to make FreeTube great.

**Comment by XanderLeaDaren** [comment]
What about a setting, allowing users to choose the default font application, and/or check a “Use system font” box?

**Comment by efb4f5ff-1298-471a-8973-3d47447115dc** [comment]
We wont implement this. If you would like this feature you should create custom build for it yourself

**Comment by Hund** [comment]
> We wont implement this. If you would like this feature you should create custom build for it yourself

You're obviously free to do whatever you want with your project. However. It feels a little bit anti-user friendly to not let the users choose how the app should look in their own environment. :)
